cmd/bacalhau/wasm_run.go

```diff
@@ -1,28 +1,83 @@
 package bacalhau
 
 import (
+	"bytes"
 	"fmt"
+	"os"
+	"path/filepath"
+	"time"
 
 	"github.com/filecoin-project/bacalhau/pkg/executor/wasm"
+	"github.com/filecoin-project/bacalhau/pkg/ipfs"
+	"github.com/filecoin-project/bacalhau/pkg/model"
 	"github.com/filecoin-project/bacalhau/pkg/system"
+	"github.com/filecoin-project/bacalhau/pkg/util/targzip"
 	"github.com/filecoin-project/bacalhau/pkg/version"
+	"github.com/ipfs/go-cid"
 	"github.com/spf13/cobra"
 	"github.com/tetratelabs/wazero"
 	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
 )
 
+var wasmJob *model.Job
+
+var runtimeSettings *RunTimeSettings
+var downloadSettings *ipfs.IPFSDownloadSettings
+
 func init() { //nolint:gochecknoinits // idiomatic for cobra commands
-	wasmCmd.AddCommand(runWasmCommand)
-	wasmCmd.AddCommand(validateWasmCommand)
+	wasmJob, _ = model.NewJobWithSaneProductionDefaults()
+	wasmJob.Spec.Engine = model.EngineWasm
+	wasmJob.Spec.Verifier = model.VerifierDeterministic
+	wasmJob.Spec.Wasm.EntryPoint = "_start"
+	wasmJob.Spec.Outputs = []model.StorageSpec{
+		{
+			Name: "outputs",
+			Path: "/outputs",
+		},
+	}
+
+	wasmCmd.AddCommand(
+		runWasmCommand,
+		validateWasmCommand,
+	)
+
+	runtimeSettings = NewRunTimeSettings()
+	settingsFlags := NewRunTimeSettingsFlags(runtimeSettings)
+	runWasmCommand.Flags().AddFlagSet(settingsFlags)
 
-	runWasmCommand.PersistentFlags().StringSliceVarP(
-		&OLR.InputUrls, "input-urls", "u", OLR.InputUrls,
+	downloadSettings = ipfs.NewIPFSDownloadSettings()
+	downloadFlags := NewIPFSDownloadFlags(downloadSettings)
+	runWasmCommand.Flags().AddFlagSet(downloadFlags)
+
+	runWasmCommand.PersistentFlags().Var(
+		VerifierFlag(&wasmJob.Spec.Verifier), "verifier",
+		`What verification engine to use to run the job`,
+	)
+	runWasmCommand.PersistentFlags().Var(
+		PublisherFlag(&wasmJob.Spec.Publisher), "publisher",
+		`What publisher engine to use to publish the job results`,
+	)
+	runWasmCommand.PersistentFlags().IntVarP(
+		&wasmJob.Deal.Concurrency, "concurrency", "c", wasmJob.Deal.Concurrency,
+		`How many nodes should run the job`,
+	)
+	runWasmCommand.PersistentFlags().IntVar(
+		&wasmJob.Deal.Confidence, "confidence", wasmJob.Deal.Confidence,
+		`The minimum number of nodes that must agree on a verification result`,
+	)
```

Comment:
Yes – I drew the line for this refactor at the job spec, because the WASM command fills up its job spec using the flags but the other jobs do it manually currently, which makes unifying it a bit more difficult. If we choose to refactor the other commands to fill up from flags too then this is much easier.